### DISCUSSION TITLE
0.4.1 Release Timeline

### DISCUSSION THREAD

**Discussion by calebrob6** [torchgeo/torchgeo#1146] [Announcements]
We're planning a 0.4.1 release in approximately a week. This will include all bug fixes in [0.4.1](https://github.com/microsoft/torchgeo/milestone/12).

If you have any bugs you would like to see fixed, now is the time to submit PRs!

**Comment by calebrob6** [comment]
# TorchGeo 0.4.1 Release Notes

This is a bugfix release. There are no new features or API changes with respect to the 0.4.0 release.

## Dependencies

Some dependencies have changed:

* nbmake: 1.3.3+ required now (#1124)
* omegaconf: now optional (#1214)
* pytorch-lightning: replaced with lightning (#1178, #1179)
* sphinx: 6+ not yet supported (#1144)
* tensorboard: now optional (#1214)
* `pip install torchgeo[all]` added, installs all optional dependencies (#1095)

Other dependencies now support newer versions:

* black: add 23 support (#1080)
* kornia: add 0.6.10 support (#1123)
* lightning: add 2 support (#1178)
* mypy: add 1 support (#1089)
* nbsphinx: add 0.9 support (#1173)
* pandas: add 2 support (#1216)
* pyvista: add 0.38 support (#1083)
* radiant-mlhub: add 0.5 support (#1102)
* scikit-image: add 0.20 support (#1153)
* setuptools: add 67 support (#1066)
* torch: add 2 support (#1177)
* torchvision: add 0.15 support (#1177)

## Datamodules

* SeCo: fix transforms (#1166)

## Datasets

Fixes for benchmark datasets:

* BigEarthNet: fix order of class labels (#1127)
* CDL: add checksum for 2022 mask (#1201)
* EuroSAT: fix SSL issue, redistribute on Hugging Face (#1065, #1072)
* FAIR1M: fix directory name (#1098, #1099)
* Landsat: better default bands (#1169)
* UC Merced: redistribute on Hugging Face (#1076)
* USAVars: fix class labels (#1138)

Fixes for base classes:

* RasterDataset: fix support for datasets where `all_bands` does not actually contain _all_ bands (e.g., Landsat) (#1134, #1135)
* RasterDataset: fix support for datasets where `all_bands` is not defined and `separate_files` is False (#1135)
* RasterDataset: fix bug when `separate_files` and no date in `filename_regex` (#1191)
* RasterDataset: remove unnecessary glob (#1219)
* RasterDataset: better error message when no data found (#1193)
* IntersectionDataset: better error message when no overlap (#1192)

## Models

There are several improvements to our new pre-trained weights:

* Add sha256 suffix for security (#1105)
* Add and improve normalizations (#1119, #1166)

## Trainers

* BYOL: Fix image size to match ViT patch size (#1084)
* Fix support for loading ViT weights (#1049, #1084)
* Fix support for non-TensorBoardLogger (#1143, #1145)

## Tests

A lot of work in this patch release went towards improving CI:

* Constrain dependencies to avoid CI hang (#1062)
* Codecov: use repository upload token (#1077)
* Cache pip installs (#1057)
* Cancel in-progress jobs on new commit (#1094) but not the labeler tasks (#1187)
* Test notebooks when they are modified (#1097)
* Speed up object detection tests (#1148)
* Fix tests on macOS arm64 (MPS support) (#1188)
* Properly test pre-trained model transforms (#1166)
* Speed up notebook tests (#665, #1124)

## Documentation

* Update the example embedded in the README (#1211)
* Fix broken URLs throughout the documentation (#1125)
* Tutorial downloads are now much smaller and faster (#1124)
* Replace CSV with TensorBoard in Trainer tutorial (#1163, #1189)
* Fix version selection button (#1144)

## Contributors

This release is thanks to the following contributors:

@adamjstewart
@ashnair1
@bugraaldal
@calebrob6
@isaaccorley 
@julien-blanchon
@lucastao
@nilsleh
@SpontaneousDuck
@TolgaAktas
